res/controllers/Denon-MC7000-scripts.js

```diff
@@ -810,11 +839,23 @@ MC7000.stopTime = function(channel, control, value, status, group) {
 MC7000.reverse = function(channel, control, value, status, group) {
     var deckNumber = script.deckFromGroup(group);
     if (value > 0) {
-        // while the button is pressed spin back
-        engine.brake(deckNumber, true, MC7000.factor[deckNumber], - 15); // start at a rate of -15 and decrease by "factor"
+        if (engine.getValue(group, "slip_enabled"))  {
+        // backspin while button is pressed at a rate of -15 and decrease by "factor"
+            engine.brake(deckNumber, value > 0, MC7000.factor[deckNumber], - 10);
+        } else {
+            engine.brake(deckNumber, true, MC7000.factor[deckNumber], - 10);
+        }
```

Comment:
this change has 0 effect as `value > 0` is always true. Moreover, you are passing the engine just the factor while the ` - 10` part is being ignored (because its passed as the 4th parameter but the function only expects 3).

Reply:
That lines are taken directly from the Mixxx Wiki -> Midi Scripting:
```
Activate brake on button press, jump to normal speed on button release (short version)

    MyControllerPrefix.brake_button = function(channel, control, value, status, group) {
        var deck = parseInt(group.substring(8,9)); // work out which deck we are using 
        engine.brake(deck, value > 0, 1.2, -10); // start at a rate of -10 and decrease at a factor of 1.2
    }
```
and works great. I can change the last value -10 and get immediate effect on the start spead for the spinback, so it is not ignored. If I delete the `value > 0` part and keep only 3 parameters then it is spinning fast forward according to the negative "Factor"

Reply:
difference between lines 844 and 846 is:
844:  if SLIP is enabled it jumps back to the original timeline upon button relese, so only backspin while button is hold down
846: if SLIP is off then the release of that button will have no effect and backspin continues until tempo = 0

Reply:
Ah, I guess I didn't look far enough in the wiki, then the first part of that wiki section seems to be outdated. You can ignore the first part.



> difference between lines 844 and 846 is:
> 844: if SLIP is enabled it jumps back to the original timeline upon button relese, so only backspin while button is hold down
> 846: if SLIP is off then the release of that button will have no effect and backspin continues until tempo = 0

Yes, that the case on button release, however, the first part of the `if (value > 0)` case only handles the button-down-case which executes the same identical code whether slip mode is engaged or not.

Reply:
I understand what you are saying but I cannot get it otherwise to work. 

The logic is and therefore I applied the `if (value > 0)` to check upon button press if slip is enabled or disabled and perform one of the 2 actions.
If the value is 0 then it makes something else - to decide either to play at the original position (slip off), runs the timer to activate slip again. 

So how would I make that without the initial check for `value > 0`

Reply:
> only handles the button-down-case which executes the same identical code whether slip mode is engaged or not

This is not quite right. On button press it decides already what the engine.brake() will do when the button is released again. either to stop or continue engine.brake() depending on the SLIP status

Reply:
I just checked for the following possibility 
```
MC7000.reverse = function(channel, control, value, status, group) {
    var deckNumber = script.deckFromGroup(group);

        if (engine.getValue(group, "slip_enabled"))  {
            // backspin while button is pressed at a rate of -10 and decrease by "factor"
            engine.brake(deckNumber, true > 0, MC7000.factor[deckNumber], - 10);
        } else {
            // without SLIP mode the backspin will continue after button release until stop
            engine.brake(deckNumber, true, MC7000.factor[deckNumber], - 10);
        }
     if (value ===0) {
        if (engine.getValue(group, "slip_enabled")) {
            engine.brake(deckNumber, false); // disable brake effect
            engine.setValue(group, "play", 1);
            engine.setValue(group, "slip_enabled", false);
            engine.beginTimer(20, function() {
                engine.setValue(group, "slip_enabled", true);
            }, true);
        } else {
            engine.softStart(deckNumber, true, MC7000.factor[deckNumber]);
        }
    }
};
```
but then it triggers backspin upon button press and button release - 2 times which is not what i want. This check is needed.

Reply:
We seem to have a misunderstanding here.

> This if (value > 0) check is needed.

Yes, absolutely, but the part I posted below can be simplified because of the preceding `if (value > 0)`:

```javascript
    if (value > 0) {
        if (engine.getValue(group, "slip_enabled"))  {
            // backspin while button is pressed at a rate of -10 and decrease by "factor"
            engine.brake(deckNumber, value > 0, MC7000.factor[deckNumber], - 10);
        } else {
            // without SLIP mode the backspin will continue after button release until stop
            engine.brake(deckNumber, true, MC7000.factor[deckNumber], - 10);
        }
```
can just be simplified to:
```javascript
    if (value > 0) {
        // while the button is pressed spin back
        // start at a rate of -10 and decrease by "factor"
        engine.brake(deckNumber, true, MC7000.factor[deckNumber], -10);
```

Reply:
Okay - quick check shows it seems to work this way. I gonna test a little and push again.
Your sentence:
`Yes, that the case on button release, however, the first part of the if (value > 0) case only handles the button-down-case which executes the same identical code whether slip mode is engaged or not.`
gave me to understand that the if statement for value check is not needed.
Thanks for clarification....

Reply:
Yes, I'm sorry for not expressing myself clearly.